drivers/sensor/i3g4250d/i3g4250d.c

```diff
@@ -0,0 +1,212 @@
+/* ST Microelectronics I3G4250D gyro driver
+ *
+ * Copyright (c) 2021 Jonathan Hahn
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Datasheet:
+ * https://www.st.com/resource/en/datasheet/i3g4250d.pdf
+ */
+
+#define DT_DRV_COMPAT st_i3g4250d
+
+#include <init.h>
+#include <sys/__assert.h>
+#include <sys/byteorder.h>
+#include <drivers/sensor.h>
+#include <logging/log.h>
+
+#include "i3g4250d.h"
+
+#define RAW_TO_MICRODEGREEPERSEC 8750
+
+LOG_MODULE_REGISTER(i3g4250d, CONFIG_SENSOR_LOG_LEVEL);
+
+static int i3g4250d_sample_fetch(const struct device *dev,
+				 enum sensor_channel chan)
+{
+
+	struct i3g4250d_data *i3g4250d = dev->data;
+	uint8_t reg;
+	int16_t buf[3] = { 0, 0, 0 };
+
+	__ASSERT_NO_MSG(chan == SENSOR_CHAN_ALL || chan == SENSOR_CHAN_GYRO_XYZ);
+
+	i3g4250d_flag_data_ready_get(i3g4250d->ctx, &reg);
+
+	if (reg) {
+		if (i3g4250d_angular_rate_raw_get(i3g4250d->ctx, buf) < 0) {
+			LOG_DBG("Failed to fetch raw data sample!");
+			return -EIO;
+		}
+		;
+
+		i3g4250d->angular_rate[0] = buf[0];
+		i3g4250d->angular_rate[1] = buf[1];
+		i3g4250d->angular_rate[2] = buf[2];
+
+		return 0;
+	} else {
+		return -EIO;
+	}
+}
+
+static inline void i3g4250d_convert(struct sensor_value *val, int16_t raw_value)
+{
+	val->val1 = (int16_t)(raw_value * RAW_TO_MICRODEGREEPERSEC / 100000LL);
+	val->val2 = (int16_t)(raw_value * RAW_TO_MICRODEGREEPERSEC) % 1000000LL;
+}
+
+static void i3g4250d_channel_convert(enum sensor_channel chan,
+				     uint16_t *raw_xyz,
+				     struct sensor_value *val)
+{
+	int i;
+	uint8_t ofs_start, ofs_stop;
+
+	switch (chan) {
+	case SENSOR_CHAN_GYRO_X:
+		ofs_start = ofs_stop = 0U;
+		break;
+	case SENSOR_CHAN_GYRO_Y:
+		ofs_start = ofs_stop = 1U;
+		break;
+	case SENSOR_CHAN_GYRO_Z:
+		ofs_start = ofs_stop = 2U;
+		break;
+	default:
+		ofs_start = 0U;
+		ofs_stop = 2U;
+		break;
+	}
+
+
+	for (i = ofs_start; i <= ofs_stop; i++) {
+		i3g4250d_convert(val++, raw_xyz[i]);
+	}
+}
+
+static int i3g4250d_channel_get(const struct device *dev,
+				enum sensor_channel chan,
+				struct sensor_value *val)
+{
+
+	struct i3g4250d_data *i3g4250d = dev->data;
+
+
+	switch (chan) {
+	case SENSOR_CHAN_GYRO_X:
+	case SENSOR_CHAN_GYRO_Y:
+	case SENSOR_CHAN_GYRO_Z:
+	case SENSOR_CHAN_GYRO_XYZ:
+		i3g4250d_channel_convert(chan, i3g4250d->angular_rate, val);
+		return 0;
+	default:
+		return -ENOTSUP;
+	}
+}
+
+static i3g4250d_dr_t gyr_odr_to_reg(const struct sensor_value *val)
+{
+	double odr = sensor_value_to_double((struct sensor_value *)val);
```

Comment:
because of the const qualifier, is there another way?

and I do not understand why sensor_value_to_double is not defined like

```
double sensor_value_to_double(const struct sensor_value *val) ..
```